cnxeasybake/oven.py

```diff
@@ -1334,3 +1340,15 @@ def copy_w_id_suffix(elem, suffix="_copy"):
     for id_elem in mycopy.xpath('//*[@id]'):
         id_elem.set('id', id_elem.get('id') + suffix)
     return mycopy
+
+
+def fetch_nodes(url):
+    """Get the contents at url, return parsed nodes child of body tag."""
+    req = requests.get(url)
+    if req.ok:
+        html = etree.HTML(req.text)
+        nodes = html.xpath('body/*')
```

Comment:
implemented all of `http://example.com/path/to/content`, `http://example.com/path/to/content#idofinterest`, and two parameter versions, taking a second param that is a CSS selector, that is applied to the result of fetching the url. The id selector and the CSS stack, if both are provided. The first case, bare URL, special cases returning just the  `body` element, converted in-place to a div (so any attributes or namespaces are preserved). Note that this means the base-url + CSS has access to the `head` parts of the page (i.e. title, keywords, etc.)